bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+
+    public static final int WEATHER_COMPANY_API_TIMEOUT = 10000;
+    public static final int REFRESH_JOB_INITIAL_DELAY = 3;
+
+    // Used to sanitize the API key in the URL in debug log messages
+    private static final String REPLACE_API_KEY = "XXXXXXXXXXXXXXXXXXXXX";
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherCompanyHandler.class);
+
+    private final Gson gson = new GsonBuilder().serializeNulls().create();
+
+    // Provided by handler factory
+    private @NonNullByDefault({}) TimeZoneProvider timeZoneProvider;
+    private @NonNullByDefault({}) HttpClient httpClient;
+    private @NonNullByDefault({}) SystemOfUnits systemOfUnits;
+
+    // Thing configuration
+    private @NonNullByDefault({}) WeatherCompanyConfig config;
+
+    private @Nullable String forecastUrl;
+    private @Nullable String pwsUrl;
+
+    // Job to update the forecast and PWS observations
+    private @Nullable Future<?> refreshJob;
+    private int refreshInterval;
+
+    private Runnable refreshRunnable = new Runnable() {
+        @Override
+        public void run() {
+            refreshForecast();
+            refreshPwsObservations();
+        }
+    };
+
+    public WeatherCompanyHandler(Thing thing, TimeZoneProvider timeZoneProvider, HttpClient httpClient,
+            UnitProvider unitProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        this.httpClient = httpClient;
+        this.systemOfUnits = unitProvider.getMeasurementSystem();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Get the configuration
+        config = getConfigAs(WeatherCompanyConfig.class);
+        // logger.debug("Handler: Config: apiKey={}", config.apiKey);
+        logger.debug("Handler: Config: locationType={}", config.locationType);
+        logger.debug("Handler: Config: postalCode={}", config.postalCode);
+        logger.debug("Handler: Config: geocode={}", config.geocode);
+        logger.debug("Handler: Config: language={}", config.language);
+        logger.debug("Handler: Config: pwsStationId={}", config.pwsStationId);
+        logger.debug("Handler: Config: refreshInterval={} minutes", config.refreshInterval);
+
+        // Set the frequency with which the forecast will be refreshed
+        refreshInterval = getRefreshIntervalSeconds();
+
+        // Construct the URL for querying the forecast
+        forecastUrl = buildForecastUrl();
+
+        // Construct the URL for querying the PWS observations
+        pwsUrl = buildPwsUrl();
+
+        // Schedule the job to refresh the forecast
+        scheduleRefreshJob();
+    }
+
+    @Override
+    public void dispose() {
+        cancelRefreshJob();
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Handler doesn't handle any commands
```

Comment:
You should handle the refresh command, optionally with some cache so that you don't overload the backend.

Reply:
Is Github right that this is still open?

Reply:
I have a solution for it. I just didn't put an update here yet.